### PR TITLE
[TEST] feat: implement nodeSelector and resources inheritance from Kubernetes resources

### PR DESCRIPTION
**Pull request by rberrelleza**
## Summary

This PR implements comprehensive Kubernetes inheritance functionality for both **resources** and **nodeSelector** settings. When these sections are omitted from the okteto.yaml manifest, Okteto can now automatically inherit the corresponding settings from the original Kubernetes PodSpec, providing a seamless development experience that respects the production resource constraints and node placement requirements.

### 🔧 Resource Inheritance
- **Environment Variable**: `OKTETO_INHERIT_KUBERNETES_RESOURCES_SETTINGS=true`
- **Functionality**: Automatically inherits CPU/memory requests and limits from the original Kubernetes container
- **Behavior**: When enabled and `resources` section is omitted, copies `resources.requests` and `resources.limits` from the target container

### 🎯 NodeSelector Inheritance  
- **Environment Variable**: `OKTETO_INHERIT_KUBERNETES_NODESELECTOR_SETTINGS=true`
- **Functionality**: Automatically inherits node placement constraints from the original Kubernetes PodSpec
- **Behavior**: When enabled and `nodeSelector` section is omitted, copies `nodeSelector` from the target PodSpec

### Helper Functions
- `ShouldInheritKubernetesResources()` - checks if resource inheritance is enabled
- `ShouldInheritKubernetesNodeSelector()` - checks if nodeSelector inheritance is enabled  
- `HasEmptyResources()` - determines if dev resources are empty (no requests/limits specified)
- `HasEmptyNodeSelector()` - determines if dev nodeSelector is empty
- `GetInheritedResourcesFromContainer()` - copies resource requirements from Kubernetes container

### Integration Points
- **GetTranslations()** function in `pkg/k8s/apps/crud.go` applies inheritance logic
- Inheritance occurs during translation rule processing for each dev container
- Resources inherited from the target container's `Resources` field
- NodeSelector inherited from the target PodSpec's `NodeSelector` field

## Behavior Matrix

| Environment Variable | Manifest Section | Behavior |
|---------------------|------------------|----------|
| `OKTETO_INHERIT_KUBERNETES_RESOURCES_SETTINGS=true` | `resources` omitted | ✅ Inherits from Kubernetes container |
| `OKTETO_INHERIT_KUBERNETES_RESOURCES_SETTINGS=true` | `resources` specified | ❌ Uses manifest values |
| `OKTETO_INHERIT_KUBERNETES_RESOURCES_SETTINGS=false` | Any | ❌ Uses manifest values or defaults |
| `OKTETO_INHERIT_KUBERNETES_NODESELECTOR_SETTINGS=true` | `nodeSelector` omitted | ✅ Inherits from Kubernetes PodSpec |
| `OKTETO_INHERIT_KUBERNETES_NODESELECTOR_SETTINGS=true` | `nodeSelector` specified | ❌ Uses manifest values |
| `OKTETO_INHERIT_KUBERNETES_NODESELECTOR_SETTINGS=false` | Any | ❌ Uses manifest values or defaults |

## Testing

✅ **Resource Inheritance Tests**: 8 comprehensive test cases covering various resource scenarios  
✅ **NodeSelector Inheritance Tests**: 13 test cases covering nodeSelector inheritance logic  
✅ **Integration Tests**: 3 test cases for GetTranslations integration  
✅ **Backwards Compatibility**: All existing tests continue to pass  
✅ **Edge Cases**: Empty resources, mixed inheritance scenarios, disabled features

## Benefits

- **🔄 Seamless Migration**: Developers don't need to manually copy resource settings from production
- **📊 Resource Consistency**: Development environments respect production resource constraints  
- **🎯 Node Placement**: Development pods can inherit production node placement requirements
- **⚡ Zero Configuration**: Works automatically when environment variables are enabled
- **🔒 Backwards Compatible**: No impact on existing configurations

## Related

- **Jira**: https://okteto.atlassian.net/browse/PROD-394
- **Feature Category**: Developer Experience Enhancement
- **Impact**: Improves development-production parity and reduces configuration overhead

---

*Developed together with Okteto AI Agent Fleets*

### CODE REVIEW

File: pkg/k8s/apps/crud_test.go [hunk 1]
```diff
@@ -543,3 +544,337 @@ func TestListDevModeOn(t *testing.T) {
...
+			assert.NotNil(t, mainTranslation)
+
+			// Check the translation rule resources
+			if len(mainTranslation.Rules) > 0 {
+				if tt.shouldInherit {
+					assert.Equal(t, tt.expectedResources, mainTranslation.Rules[0].Resources)
+				} else {
+					// When not inheriting, the rule should have the original dev resources
+					assert.Equal(t, tt.dev.Resources, mainTranslation.Rules[0].Resources)
+				}
```
Comment (ifbyol) [inline pkg/k8s/apps/crud_test.go]: @jLopezbarb Let's split this test to prevent these conditionals. They make harder to read and maintain this test

File: pkg/k8s/apps/crud_test.go [hunk 2]
```diff
@@ -543,3 +544,337 @@ func TestListDevModeOn(t *testing.T) {
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Save original environment variable value
+			originalValue := os.Getenv(model.OktetoInheritKubernetesResourcesEnvVar)
+			defer func() {
+				if originalValue == "" {
+					os.Unsetenv(model.OktetoInheritKubernetesResourcesEnvVar)
+				} else {
+					os.Setenv(model.OktetoInheritKubernetesResourcesEnvVar, originalValue)
+				}
```
Comment (ifbyol) [inline pkg/k8s/apps/crud_test.go]: Let's use `t.setenv`, so we don't have to explicitly clean it up. Also, let's try to remove the if/else

File: pkg/k8s/apps/crud_test.go [hunk 3]
```diff
@@ -543,3 +544,337 @@ func TestListDevModeOn(t *testing.T) {
...
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Set environment variable
+			if tt.envVarValue != "" {
+				os.Setenv(model.OktetoInheritKubernetesNodeSelectorEnvVar, tt.envVarValue)
+			} else {
+				os.Unsetenv(model.OktetoInheritKubernetesNodeSelectorEnvVar)
+			}
+			defer os.Unsetenv(model.OktetoInheritKubernetesNodeSelectorEnvVar)
```
Comment (ifbyol) [inline pkg/k8s/apps/crud_test.go]: Let's use t.setenv, so we don't have to explicitly clean it up. Also, let's try to remove the if/else

File: pkg/k8s/apps/translate_test.go [hunk 4]
```diff
@@ -2306,3 +2306,110 @@ func TestTranslateLifecycle(t *testing.T) {
 		})
 	}
 }
+
+func TestDev_InheritResourcesFromContainer(t *testing.T) {
```
Comment (ifbyol) [inline pkg/k8s/apps/translate_test.go]:
```suggestion
func TestDev_GetInheritedResourcesFromContainer(t *testing.T) {
```

File: pkg/model/dev_test.go [hunk 5]
```diff
@@ -1488,3 +1477,218 @@ func TestPrepare(t *testing.T) {
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Save original value
+			originalValue := os.Getenv(OktetoInheritKubernetesResourcesEnvVar)
+			defer func() {
+				if originalValue == "" {
+					os.Unsetenv(OktetoInheritKubernetesResourcesEnvVar)
+				} else {
+					os.Setenv(OktetoInheritKubernetesResourcesEnvVar, originalValue)
+				}
```
Comment (ifbyol) [inline pkg/model/dev_test.go]: Let's use `t.Setenv`

File: pkg/model/dev_test.go [hunk 6]
```diff
@@ -1488,3 +1477,218 @@ func TestPrepare(t *testing.T) {
...
+			}()
+
+			// Set test value
+			if tt.envValue == "" {
+				os.Unsetenv(OktetoInheritKubernetesResourcesEnvVar)
+			} else {
+				os.Setenv(OktetoInheritKubernetesResourcesEnvVar, tt.envValue)
+			}
```
Comment (ifbyol) [inline pkg/model/dev_test.go]: Same here about `t.Setenv`

File: pkg/model/dev_test.go [hunk 7]
```diff
@@ -1488,3 +1477,218 @@ func TestPrepare(t *testing.T) {
...
+
+			result := env.LoadBooleanOrDefault(OktetoInheritKubernetesResourcesEnvVar, false)
```
Comment (ifbyol) [inline pkg/model/dev_test.go]: I don't think this test provides value. We are testing `env.LoadBooleanOrDefault` function, which is not related to our code. It is a function from the standard library. I would just remove this test

File: pkg/model/dev_test.go [hunk 8]
```diff
@@ -1488,3 +1477,218 @@ func TestPrepare(t *testing.T) {
...
+		t.Run(tt.name, func(t *testing.T) {
+			// Set environment variable
+			if tt.envValue != "" {
+				os.Setenv(OktetoInheritKubernetesNodeSelectorEnvVar, tt.envValue)
+			} else {
+				os.Unsetenv(OktetoInheritKubernetesNodeSelectorEnvVar)
+			}
+			defer os.Unsetenv(OktetoInheritKubernetesNodeSelectorEnvVar)
+
+			result := env.LoadBooleanOrDefault(OktetoInheritKubernetesNodeSelectorEnvVar, false)
```
Comment (ifbyol) [inline pkg/model/dev_test.go]: Another test testing an standard library function not related with our code. I would just delete this test

File: pkg/model/dev_test.go [hunk 9]
```diff
@@ -26,8 +26,108 @@ import (
 	"github.com/spf13/afero"
 	"github.com/stretchr/testify/assert"
 	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 )
 
+// Test helpers
+func assertDevFields(t *testing.T, dev, main *Dev) {
+	t.Helper()
+	if dev.Name != "deployment" {
```
Comment (ifbyol) [inline pkg/model/dev_test.go]: Nit: Would it make sense to take the chance to change the asserts to use `testify` library?

**Review by ifbyol — COMMENTED** [review]
Functionality looks good, but there are several things I would address before merging

**Review by ifbyol — APPROVED** [review]
It is looking better after the changes in the tests! 👏
